src/web/src/components/SignUp/Forms/GitHubAccount.tsx

```diff
@@ -0,0 +1,191 @@
+  createStyles({
+    root: {
+      padding: '0',
+      margin: '0',
+      width: '100%',
+      position: 'relative',
+      minHeight: '100%',
+    },
+    container: {
+      display: 'grid',
+      gridTemplateAreas: '1fr',
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'start',
+      width: '100%',
+      position: 'absolute',
+      minHeight: '100%',
+      [theme.breakpoints.down(600)]: {
+        width: '90%',
+        marginLeft: '5%',
+      },
+    },
+    titlePage: {
+      fontSize: '1.5em',
+    },
+    subtitlePage: {
+      fontSize: '1.1em',
+      lineHeight: '1.8em',
+    },
+    infoContainer: {
+      display: 'grid',
+      gridTemplateColumns: '65% 35%',
+      gridGap: '1%',
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'center',
+      width: '90%',
+      [theme.breakpoints.down(600)]: {
+        gridTemplateColumns: '1fr',
+      },
+    },
+    inputsContainer: {
+      width: '100%',
+      display: 'grid',
+      gridTemplateColumns: '80% 20%',
+      '& .MuiFormHelperText-root': {
+        fontSize: '0.9em',
+        color: 'black',
+      },
+      '& .MuiFormLabel-root': {
+        color: 'black',
+      },
+    },
+    button: {
+      height: '35px',
+      width: '50%',
+      alignSelf: 'center',
+      fontSize: '0.8em',
+      background: '#121D59',
+      color: '#A0D1FB',
+      marginLeft: '5%',
+      '&:hover': {
+        color: 'black',
+        border: '1px solid #121D59',
+      },
+      '&.Mui-disabled': {
+        backgroundColor: 'inherit',
+      },
+    },
+    avatarPreview: {
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'center',
+      justifySelf: 'end',
+      padding: '6%',
+      borderRadius: '5px',
+      [theme.breakpoints.down(600)]: {
+        justifySelf: 'center',
+        padding: '3%',
+        marginTop: '5%',
+      },
+    },
+    username: {
+      fontSize: '1.1em',
+    },
+  })
+);
+
+const gitHubApiUrl = 'https://api.github.com/users';
+
+const GitHubAccount = connect<{}, SignUpForm>((props) => {
+  const classes = useStyles();
+  const { values, setFieldValue } = props.formik;
+  const [validating, setValidating] = useState(false);
+  const [error, setError] = useState('');
+  const controllerRef = useRef<AbortController | null>();
+
+  const validateGit = async () => {
+    if (!values.githubUsername) {
+      setError('');
+      setFieldValue('github', {}, true);
+      return;
+    }
+    try {
+      setValidating(true);
+      if (controllerRef.current) {
+        controllerRef.current.abort();
+      }
+      controllerRef.current = new AbortController();
+      const response = await fetch(`${gitHubApiUrl}/${values.githubUsername}`, {
+        signal: controllerRef.current?.signal,
+      });
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const res = await response.json();
```

Comment:
Nit: `res` isn't a great name here, since we often use it to mean response.  I think you might mean `result`, which would be a better choice.

Reply:
Done